test/integration/job/job_test.go

```diff
@@ -2061,20 +2079,32 @@ func TestManagedBy(t *testing.T) {
 			},
 		},
 	}
+
+	i := 0
+	closeFn, restConfig, clientSet, ns := setup(t, "managed-by")
+	t.Cleanup(closeFn)
 	for name, test := range testCases {
 		t.Run(name, func(t *testing.T) {
 			resetMetrics()
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobManagedBy, test.enableJobManagedBy)
 
-			closeFn, restConfig, clientSet, ns := setup(t, "managed-by")
-			defer closeFn()
 			ctx, cancel := startJobControllerAndWaitForCaches(t, restConfig)
 			defer cancel()
+
+			test.job.Name = fmt.Sprintf("test-job-%d", i)
 			jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &test.job)
 			if err != nil {
 				t.Fatalf("Error %v while creating the job %q", err, klog.KObj(jobObj))
 			}
 
+			defer func() {
+				removePodsFinalizersByName(ctx, t, clientSet, ns.Name, []string{blockDeletionFinalizerForTest, batchv1.JobTrackingFinalizer})
```

Comment:
might be worth passing the finalizers straight to the cleanUp function

Reply:
I kept the finalizers removal in a separated function because some tests where already doing this (see [this example](https://github.com/kubernetes/kubernetes/pull/126439/files#diff-ace35385ec6f5400e8ccbfe8e9f35c2f9573d0fc05889071dc9c4b36e5e975f7R1544)). And since they were running fine, I didn't want to add extra unnecessary calls.

However, if you prefer so, I can implement your suggestion and see how it all behaves.

Reply:
I think it would be more readable that way. We can pass the finalizers to the cleanup function. 

> (see [this example](https://github.com/kubernetes/kubernetes/pull/126439/files#diff-ace35385ec6f5400e8ccbfe8e9f35c2f9573d0fc05889071dc9c4b36e5e975f7R1544)).

Can we also call cleanup in this place? this way we do not have to check how many and which functions we have to call to do a proper cleanup in each place.

Reply:
@atiratree I implemented your suggestions, please have a look whenever you have time 😊

Reply:
nice!